addons/editor-number-arrow-keys/userscript.js

```diff
@@ -0,0 +1,62 @@
+export default async function ({ addon }) {
+  const settings = {
+    none: 0,
+    tenth: 0.1,
+    one: 1,
+    ten: 10,
+  };
+
+  const amountOfDecimals = (num) => {
+    if (num % 1 === 0) return 0;
+    return num.toString().split(".")[1].length;
+  };
+
+  const shiftDecimalPointToRight = (num, times) => {
+    const numberIsNegative = num[0] === "-";
+    let numStr = num.toString();
+    if (numberIsNegative) numStr = numStr.substring(1);
+    for (let i = 0; i < times; i++) {
+      if (numStr.indexOf(".") === -1) numStr += 0;
+      else if (numStr.indexOf(".") === numStr.length - 2) numStr = numStr.replace(".", "");
+      else {
+        const index = numStr.indexOf(".");
+        const numArrFiltered = Array.from(numStr.replace(".", ""));
+        numArrFiltered.splice(index + 1, 0, ".");
+        numStr = numArrFiltered.join("");
+      }
+    }
+    return Number(numStr) * (numberIsNegative ? -1 : 1);
+  };
+
+  document.body.addEventListener("keydown", (e) => {
+    if (!e.target.classList.contains("blocklyHtmlInput")) return;
+    if (!["ArrowUp", "ArrowDown"].includes(e.code)) return;
+    if (Number(e.target.value).toString().replace(/^0*/, "") !== e.target.value.replace(/^0*/, "")) return;
+    // TODO: why remove leading zeros to a result of Number().toString ?
```

Comment:
Because `e.target.value` might be `.1`, where `Number(e.target.value).toString()` would be "0.1", and therefore different from `e.target.value` even though they should be counted as the same.

```suggestion
```